### PR TITLE
feat: correct homonymadditons api & lambda

### PR DESCRIPTION
**Pull request by emalfroy**

### CODE REVIEW

File: src/StreetNameRegistry.Api.BackOffice.Handlers.Lambda/MessageHandler.cs [hunk 1]
```diff
@@ -65,6 +65,10 @@ public async Task HandleMessage(object? messageData, MessageMetadata messageMeta
                     await mediator.Send(new CorrectStreetNameRetirementLambdaRequest(messageMetadata.MessageGroupId!, request), cancellationToken);
                     break;
 
+                case CorrectStreetNameHomonymAdditionsSqsRequest request:
```
Comment (jvandaal) [inline src/StreetNameRegistry.Api.BackOffice.Handlers.Lambda/MessageHandler.cs:68]: Missing test.

File: src/StreetNameRegistry/Municipality/Municipality_StreetName.cs [hunk 2]
```diff
@@ -143,18 +143,20 @@ public void CorrectStreetNameRetirement(PersistentLocalId persistentLocalId)
...
+            PersistentLocalId persistentLocalId,
+            HomonymAdditions homonymAdditions)
         {
             StreetNames
                 .GetNotRemovedByPersistentLocalId(persistentLocalId)
                 .CorrectHomonymAdditions(homonymAdditions, GuardUniqueActiveStreetNameNames);
         }
 
-        public void RemoveStreetNameHomonymAdditions(PersistentLocalId persistentLocalId, List<Language> languages)
+        public void RemoveStreetNameHomonymAdditions(PersistentLocalId persistentLocalId, List<Language> homonymsToRemove)
```
Comment (jvandaal) [inline src/StreetNameRegistry/Municipality/Municipality_StreetName.cs]: Undo rename.

File: src/StreetNameRegistry.Api.BackOffice/StreetNameController-CorrectHomonymAdditions.cs [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+            }
+            catch (AggregateNotFoundException)
+            {
+                throw new ApiException(ValidationErrors.Common.StreetNameNotFound.Message, StatusCodes.Status404NotFound);
+            }
+            catch (StreetNameIsNotFoundException)
+            {
+                throw new ApiException(ValidationErrors.Common.StreetNameNotFound.Message, StatusCodes.Status404NotFound);
+            }
+            catch (HomonymAdditionMaxCharacterLengthExceededException e)
```
Comment (jvandaal) [inline src/StreetNameRegistry.Api.BackOffice/StreetNameController-CorrectHomonymAdditions.cs]: This should be checked in the validator, but the validator will not throw this exception.

File: test/StreetNameRegistry.Tests/BackOffice/Api/WhenCorrectingStreetNameHomonymAdditions/GivenHomonymAdditionExceedsMaxLength.cs [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    public sealed class GivenHomonymAdditionExceedsMaxLength : BackOfficeApiTest<StreetNameController>
+    {
+        // TODO: FIX
+        public GivenHomonymAdditionExceedsMaxLength(ITestOutputHelper testOutputHelper) : base(testOutputHelper)
+        { }
+
+        [Fact]
+        public void ThenThrowsHomonymAdditionMaxCharacterLengthExceededException()
+        {
+            var mockRequestValidator = new Mock<IValidator<CorrectStreetNameHomonymAdditionsRequest>>();
```
Comment (jvandaal) [inline test/StreetNameRegistry.Tests/BackOffice/Api/WhenCorrectingStreetNameHomonymAdditions/GivenHomonymAdditionExceedsMaxLength.cs]: The Validator doesn't throw domain exceptions.

File: src/StreetNameRegistry.Projections.Legacy/StreetNameListV2/StreetNameListProjectionsV2.cs [hunk 5]
```diff
@@ -136,6 +136,42 @@ await context
                     }, ct);
             });
 
+            When<Envelope<StreetNameHomonymAdditionsWereCorrected>>(async (context, message, ct) =>
```
Comment (jvandaal) [inline src/StreetNameRegistry.Projections.Legacy/StreetNameListV2/StreetNameListProjectionsV2.cs:139]: @ArneD should the event also be handled in StreetNameNameProjectionsV2? This projection doesn't know homonym additions, but maybe `UpdateVersionTimestamp` is necessary?

File: src/StreetNameRegistry.Api.BackOffice/StreetNameController-CorrectHomonymAdditions.cs [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+        [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status412PreconditionFailed)]
+        [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status500InternalServerError)]
+        [SwaggerResponseHeader(StatusCodes.Status202Accepted, "location", "string", "De URL van het aangemaakte ticket.")]
+        [SwaggerResponseExample(StatusCodes.Status400BadRequest, typeof(BadRequestResponseExamples))]
+        [SwaggerResponseExample(StatusCodes.Status412PreconditionFailed, typeof(PreconditionFailedResponseExamples))]
+        [SwaggerResponseExample(StatusCodes.Status500InternalServerError, typeof(InternalServerErrorResponseExamples))]
+        [Authorize(AuthenticationSchemes = JwtBearerDefaults.AuthenticationScheme, Policy = PolicyNames.Adres.DecentraleBijwerker)]
+        [Authorize(AuthenticationSchemes = JwtBearerDefaults.AuthenticationScheme, Policy = PolicyNames.Adres.InterneBijwerker)]
+        public async Task<IActionResult> CorrectHomonymAddition(
```
Comment (ArneD) [inline src/StreetNameRegistry.Api.BackOffice/StreetNameController-CorrectHomonymAdditions.cs]: plural?

**Review by ArneD — CHANGES_REQUESTED** [review]

File: src/StreetNameRegistry.Projections.Legacy/StreetNameListV2/StreetNameListProjectionsV2.cs (hunk 5 above)
Comment (ArneD) [reply src/StreetNameRegistry.Projections.Legacy/StreetNameListV2/StreetNameListProjectionsV2.cs:139]: Yes, same in `StreetNameNameProjectionsV2`

**Review by ArneD — APPROVED** [review]

**Pull request merged**